api/v1alpha4/virtualmachine_types.go

```diff
@@ -457,6 +457,88 @@ type VirtualMachineCryptoSpec struct {
 	UseDefaultKeyProvider *bool `json:"useDefaultKeyProvider,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=disk;network;cdrom
+
+// VirtualMachineBootOptionsBootableDevice represents the type of bootable device
+// that a VM may be booted from.
+type VirtualMachineBootOptionsBootableDevice string
+
+const (
+	VirtualMachineBootOptionsBootableDiskDevice    VirtualMachineBootOptionsBootableDevice = "disk"
+	VirtualMachineBootOptionsBootableNetworkDevice VirtualMachineBootOptionsBootableDevice = "network"
+	VirtualMachineBootOptionsBootableCDRomDevice   VirtualMachineBootOptionsBootableDevice = "cdrom"
+)
+
+// +kubebuilder:validation:Enum=ipv4;ipv6
+
+// VirtualMachineBootOptionsNetworkBootProtocol represents the protocol to
+// use during PXE network boot or NetBoot.
+type VirtualMachineBootOptionsNetworkBootProtocol string
+
+const (
+	VirtualMachineBootOptionsNetworkBootProtocolIPv4 VirtualMachineBootOptionsNetworkBootProtocol = "ipv4"
```

Comment:
Should be `IP4` and `IP6` to be consistent with how we do not use the `v` in the other locations in our spec either.